runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java

```diff
@@ -431,6 +431,18 @@ else if (config.swapHomePortal() != HouseMode.ENTER && option.equals("enter"))
 					break;
 			}
 		}
+		else if (config.swapTeleportToDestination() != ObeliskMode.ACTIVATE && target.equals("obelisk"))
```

Comment:
Don't check `config.swapTeleportToDestination() != ObeliskMode.ACTIVATE`, it's safe to assume swapping should always be attempted if an obelisk is clicked.